docs/_spec/05-classes-and-objects.md

```diff
@@ -801,3 +803,352 @@ Generally, a _companion module_ of a class is an object which has the same name
 Conversely, the class is called the _companion class_ of the module.
 
 Very much like a concrete class definition, an object definition may still contain declarations of abstract type members, but not of abstract term members.
+
+## Enum Definitions
+
+<!-- TODO: Agree with NTs of rest of spec -->
+```ebnf
+TmplDef   ::=  ‘enum’ EnumDef
+EnumDef   ::=  id ClassConstr [‘extends’ [ConstrApps]] EnumBody
+EnumBody  ::=  [nl] ‘{’ [SelfType] EnumStat {semi EnumStat} ‘}’
+EnumStat  ::=  TemplateStat
+            |  {Annotation [nl]} {Modifier} EnumCase
+EnumCase  ::=  ‘case’ (id ClassConstr [‘extends’ ConstrApps] | ids)
+```
+
+An _enum definition_ implies the definition of an _enum class_, a companion object, and one or more _enum cases_.
+
+Enum definitions are useful to encode both Generalised Algebraic Data Types and Enumerated Types.
+
+The compiler expands enum definitions to code that only uses
+Scala's other language features. As such, enum definitions in Scala are
+convenient _syntactic sugar_, but they are not essential to understand
+Scala's core.
```

Comment:
Please use one sentence per "physical" line, as described in the README.
```suggestion
The compiler expands enum definitions to code that only uses Scala's other language features.
As such, enum definitions in Scala are convenient _syntactic sugar_, but they are not essential to understand Scala's core.
```
This applies everywhere.

Reply:
done